starboard/loader_app/slot_management_test.cc

```diff
@@ -468,6 +489,81 @@ TEST_P(SlotManagementTest, BadSabi) {
   SbFileDeleteRecursive(good_path.c_str(), false);
 }
 
+TEST_P(SlotManagementTest, CompareEvergreenVersion) {
+  if (!storage_path_implemented_) {
+    return;
+  }
+  std::vector<char> v1(kTestEvergreenVersion1,
+                       kTestEvergreenVersion1 + strlen(kTestEvergreenVersion1));
+  std::vector<char> v2(kTestEvergreenVersion2,
+                       kTestEvergreenVersion2 + strlen(kTestEvergreenVersion2));
+  std::vector<char> v3(kMaxEgVersionLength);
+  ASSERT_EQ(0, CompareEvergreenVersion(&v1, &v3));
+  ASSERT_EQ(0, CompareEvergreenVersion(&v1, &v1));
+  ASSERT_EQ(-1, CompareEvergreenVersion(&v1, &v2));
+  v3.assign(kTestEvergreenVersion3,
+            kTestEvergreenVersion3 + strlen(kTestEvergreenVersion3));
+  ASSERT_EQ(1, CompareEvergreenVersion(&v3, &v2))
```

Comment:
We should try to write unit tests that exercise the public APIs  instead of the private helpers: go/unit-testing-practices#public-apis.

I think we can try to do this later in a follow-up PR, though.

Reply:
A follow-up PR SGTM.